workspace.go

```diff
@@ -0,0 +1,66 @@
+package twilio
+
+import (
+	"context"
+	"net/url"
+)
+
+const WorkspacePath = "Workspaces/"
+const WorkspacePathWithoutSlash = "Workspaces"
+
+type WorkspaceCreator struct {
+	client *Client
+}
+
+type Workspace struct {
+	AccountSid           string            `json:"account_sid"`
+	DateCreated          TwilioTime        `json:"date_created"`
+	DateUpdated          TwilioTime        `json:"date_updated"`
+	DefaultActivityName  string            `json:"default_activity_name"`
+	DefaultActivitySid   string            `json:"default_activity_sid"`
+	EventCallbackUrl     string            `json:"event_callback_url"`
+	EventsFilter         string            `json:"events_filter"`
+	FriendlyName         string            `json:"friendly_name"`
+	Links                map[string]string `json:"links"`
+	MultiTaskEnabled     bool              `json:"multi_task_enabled"`
+	PrioritizeQueueOrder string            `json:"prioritize_queue_order"`
+	Sid                  string            `json:"sid"`
+	TimeoutActivityName  string            `json:"timeout_activity_name"`
+	TimeoutActivitySid   string            `json:"timeout_activity_sid"`
+	URL                  string            `json:"url"`
+}
+
+// Get retrieves a Workspace by its sid.
+//
+// See https://www.twilio.com/docs/taskrouter/api/workspaces#action-create for more.
+func (r *WorkspaceCreator) Get(ctx context.Context, sid string) (*Workspace, error) {
+	workspace := new(Workspace)
+	err := r.client.WorkspaceClient.GetResource(ctx, WorkspacePathWithoutSlash, sid, workspace)
+	return workspace, err
+}
+
+// Create creates a new Workspace.
+//
+// For a list of valid parameters see
+// https://www.twilio.com/docs/taskrouter/api/workspaces#action-create
+func (r *WorkspaceCreator) Create(ctx context.Context, data url.Values) (*Workspace, error) {
+	workspace := new(Workspace)
+	err := r.client.WorkspaceClient.CreateResource(ctx, WorkspacePath, data, workspace)
+	return workspace, err
+}
+
+// Delete deletes a Workspace.
+//
+// See https://www.twilio.com/docs/taskrouter/api/workers#action-delete for more.
+func (r *WorkspaceCreator) Delete(ctx context.Context, sid string) error {
+	return r.client.WorkspaceClient.DeleteResource(ctx, WorkspacePathWithoutSlash, sid)
+}
+
+// Update updates a Workspace.
+//
+// See https://www.twilio.com/docs/taskrouter/api/workspaces#action-update for more.
+func (r *WorkspaceCreator) Update(ctx context.Context, sid string, data url.Values) (*Workspace, error) {
+	worker := new(Workspace)
+	err := r.client.WorkspaceClient.UpdateResource(ctx, WorkspacePathWithoutSlash, sid, data, worker)
+	return worker, err
+}
```

Comment:
We should probably add the `PageIterator` related function calls here as well. Just for parity with the other go files in here. Trying to minimize the work we have to do when going back to clean this up for actual merge in the original repo 😓

Reply:
The twilio docs are very confusing to me about what getting a workspace actually does. It says it returns the workers? https://www.twilio.com/docs/flex/routing/api/workspace?code-sample=code-fetch-workspace&code-language=curl&code-sdk-version=json

I don't have much context on this. it seems like there is some other go code that does this same thing. In any event, I think you are right but I don't want to take this on since I am unsure if there is a way to page between workspaces.